tm-monitor/rpc.go

```diff
@@ -37,23 +42,41 @@ func routes(m *monitor.Monitor) map[string]*rpc.RPCFunc {
 
 // RPCStatus returns common statistics for the network and statistics per node.
 func RPCStatus(m *monitor.Monitor) interface{} {
-	return func() (networkAndNodes, error) {
-		return networkAndNodes{m.Network, m.Nodes}, nil
+	return func() (NetworkAndNodes, error) {
+
+		status := NewNetworkAndNodes()
+		status.SetNetworkStatus(m)
+		status.SetNodesStatus(m)
+
+		return *status, nil
 	}
 }
 
 // RPCNetworkStatus returns common statistics for the network.
 func RPCNetworkStatus(m *monitor.Monitor) interface{} {
-	return func() (*monitor.Network, error) {
-		return m.Network, nil
+	return func() (*NetworkStatus, error) {
+
+		status := NewNetworkAndNodes()
+		status.SetNetworkStatus(m)
+
+		return status.Network, nil
 	}
 }
 
 // RPCNodeStatus returns statistics for the given node.
 func RPCNodeStatus(m *monitor.Monitor) interface{} {
-	return func(name string) (*monitor.Node, error) {
+	return func(name string) (*NodeStatus, error) {
+
 		if i, n := m.NodeByName(name); i != -1 {
-			return n, nil
+
+			status := &NodeStatus{}
```

Comment:
there is no need in this additional data structures. you can just add `amino:"unsafe"` tag to the float fields https://github.com/tendermint/tools/pull/98#discussion_r193399934

Reply:
Yeah, you are right and I have verified it. So currently could we replace wire:"unsafe" with amino:"unsafe"?

Reply:
yes